Add validation for some null arguments and dangling render passes

kainino0x opened the pull request:
And add tests for both

kainino0x commented:
I'm pretty sure there are a bunch more null arguments checks that need to be added in many places, but this is a start.

Kangz commented on src/tests/unittests/validation/CommandBufferValidationTests.cpp:
```diff
@@ -0,0 +1,62 @@
+class CommandBufferValidationTest : public ValidationTest {
+};
+
+TEST_F(CommandBufferValidationTest, Empty) {
```
nit: comment before the tests (ANGLE-style).

Kangz commented:
Null arguments are not allowed anywhere in the API currently and in fact the wire doesn't support them. I'll refactor this validation to but it in the autogenerated validation in templates/BackendProcTable.cpp.

Also I'm not sure how to split test file, for now I'm thinking of splitting them depending on which builder will generate the error, and maybe have several files for the command buffer builder if needed. What do you think?

kainino0x commented:
null args: Okay, I was wondering about that. I've been having issues where if an invalid object is passed to a builder, it shows up on the validation side as a null pointer. I'll add comments and land, but you may want to remove these extra null checks once you add it to BackendProcTable.

split: Yes, I think that's the right split. That is (or was intended to be) how I split these files.

Kangz commented:
Yep the split was fine, I was just clarifying for myself :)
